From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: kenway214 <kenway214@outlook.com>
Date: Tue, 30 Jun 2026 08:23:36 +0530
Subject: [PATCH] sepolicy: Introduce obscura [3/3]

Change-Id: Iba074d64905001d63592a49aed20831d15d569d9
Signed-off-by: kenway214 <kenway214@outlook.com>
(cherry picked from commit f5f655e5da34cc7ea4faf1dc3ec969f5114e7901)
---
 sepolicy/private/platform_app.te  | 7 +++++++
 sepolicy/private/service.te       | 3 +++
 sepolicy/private/service_contexts | 3 +++
 sepolicy/private/system_app.te    | 3 +++
 sepolicy/private/system_server.te | 3 +++
 5 files changed, 19 insertions(+)

diff --git a/sepolicy/private/platform_app.te b/sepolicy/private/platform_app.te
index 38fa551e..c452a2ab 100644
--- a/sepolicy/private/platform_app.te
+++ b/sepolicy/private/platform_app.te
@@ -5,3 +5,10 @@ hal_client_domain(platform_app, hal_aospa_powershare)
 
 # Allow LineageHealth HAL service to be found
 hal_client_domain(platform_app, hal_lineage_health)
+
+# Obscura
+allow platform_app obscura_service:service_manager find;
+allow platform_app system_data_file:dir r_dir_perms;
+allow platform_app system_data_file:file r_file_perms;
+allow platform_app system_userdir_file:dir r_dir_perms;
+allow platform_app system_userdir_file:file r_file_perms;
diff --git a/sepolicy/private/service.te b/sepolicy/private/service.te
index 94feed8b..71f1a3f8 100644
--- a/sepolicy/private/service.te
+++ b/sepolicy/private/service.te
@@ -9,3 +9,6 @@ type lineage_health_interface_service, system_api_service, system_server_service
 
 # PIH
 type pih_manager_service, app_api_service, system_api_service, system_server_service, service_manager_type;
+
+# Obscura
+type obscura_service, system_api_service, system_server_service, service_manager_type;
diff --git a/sepolicy/private/service_contexts b/sepolicy/private/service_contexts
index 4ad6a699..07af3c1d 100644
--- a/sepolicy/private/service_contexts
+++ b/sepolicy/private/service_contexts
@@ -12,3 +12,6 @@ lineagehealth                               u:object_r:lineage_health_interface_
 
 # PIH
 pih_manager                                 u:object_r:pih_manager_service:s0
+
+# Obscura
+obscura                                   u:object_r:obscura_service:s0
diff --git a/sepolicy/private/system_app.te b/sepolicy/private/system_app.te
index b50980b6..b71b9a20 100644
--- a/sepolicy/private/system_app.te
+++ b/sepolicy/private/system_app.te
@@ -1,2 +1,5 @@
 # Allow system_app to read the AOSPA version.
 get_prop(system_app, aospa_build_prop)
+
+# Obscura
+allow system_app obscura_service:service_manager find;
diff --git a/sepolicy/private/system_server.te b/sepolicy/private/system_server.te
index e8275712..9279bc79 100644
--- a/sepolicy/private/system_server.te
+++ b/sepolicy/private/system_server.te
@@ -21,3 +21,6 @@ allow system_server zygote_tmpfs:file rw_file_perms;
 
 # PIH
 add_service(system_server, pih_manager_service);
+
+# Obscura
+add_service(system_server, obscura_service);
